### PR TITLE
docs: manual deployment with asgi servers

### LINKED ISSUE #2609: Docs: Add deployment documentation
### Summary

Create documentation around deployment of Litestar applications.

This would include deployment via

- [x] Manually via `uvicorn`, `hypercorn`, `daphne`, `granian`, etc.
	- [x] https://github.com/litestar-org/litestar/pull/2663
- [ ] Baremetal/VM with something like `systemd` unit files
- [x] Via `supervisor` https://github.com/litestar-org/litestar/pull/2724/files
- [ ] Containers/LXC
- [ ] K8s
- [ ] Behind a proxy (NGINX/Traefik/HaProxy/etc)
  - [x] ``nginx-unit`` (Done in https://github.com/litestar-org/litestar/pull/2615)
- [ ] Deployment through various cloud providers

Any other deployment mechanism that may be beneficial.

## It should probably go in the `Topics` section

### PR DESCRIPTION
**Pull request by leejayhsu**
### Pull Request Checklist

- [x] New code has 100% test coverage
- [x] (If applicable) The prose documentation has been updated to reflect the changes introduced by this PR
- [ ] (If applicable) The reference documentation has been updated to reflect the changes introduced by this PR
- [x] Pre-Commit Checks were ran and passed
- [x] Tests were ran and passed

### Description
- Add a page in the Deployments section for manual deployment with ASGI servers

### Close Issue(s)
- Part of #2609

### CODE REVIEW

**Review by Alc-Alc — COMMENTED** [review]
~~Would it be better to add these as "group tabs" https://sphinx-tabs.readthedocs.io/en/latest/#group-tabs ? Just a nit :)~~ 

Sorry, I thought it was using the same extension

File: docs/topics/deployment/manually-with-asgi-server.rst [hunk 1]
```diff
@@ -0,0 +1,102 @@
...
+
+Daphne
+------
+
+`Daphne <https://github.com/django/daphne/>`_ is an ASGI server that was originally developed for `Django Channels <https://channels.readthedocs.io/en/latest/>`_, and supports HTTP/1.1, HTTP/2, and WebSocket.
+
+Granian
+-------
+
+`Granian <https://github.com/emmett-framework/granian/>`_ is a Rust based ASGI server that supports HTTP/1.1, HTTP/2, and WebSocket.
```
Comment (JacobCoffee) [inline docs/topics/deployment/manually-with-asgi-server.rst:24]: Lets put these into a tab group as well - note the following suggestions to add though

File: docs/topics/deployment/manually-with-asgi-server.rst [hunk 2]
```diff
@@ -0,0 +1,102 @@
...
+
+        .. code-block:: shell
+
+            pip install daphne
+
+    .. tab-item:: Granian
+
+        .. code-block:: shell
+
+            pip install granian
```
Comment (JacobCoffee) [inline docs/topics/deployment/manually-with-asgi-server.rst:52]: These and all following and previous tab sets can use the sync to change each tab to the value

- https://sphinx-design.readthedocs.io/en/latest/tabs.html#synchronised-tabs if you are using sphinx-design (preffered)
- https://sphinx-tabs.readthedocs.io/en/latest/#group-tabs if you are using sphinx-tabs (should be removed as a dep. tbh)

**Review by JacobCoffee — CHANGES_REQUESTED** [review]

**Comment by JacobCoffee** [comment]
Just now saw @Alc-Alc's comment hehe.. i saw the strikethrough and stopped reading. But yeah, sync tabs would be awesome :)

**Comment by JacobCoffee** [comment]
@all-contributors add @leejayhsu to docs

File: docs/topics/deployment/manually-with-asgi-server.rst (hunk 2 above)
Comment (leejayhsu) [reply docs/topics/deployment/manually-with-asgi-server.rst]: thank you for the suggestion! 🙏 
done in https://github.com/litestar-org/litestar/pull/2663/commits/e97a9ea00bd2311a793531b49c8839423b41772f

File: docs/topics/deployment/manually-with-asgi-server.rst (hunk 1 above)
Comment (leejayhsu) [reply docs/topics/deployment/manually-with-asgi-server.rst]: great suggestion! done in https://github.com/litestar-org/litestar/pull/2663/commits/188600113e06a505bdd476235d1f08480ace2616

**Review by JacobCoffee — APPROVED** [review]

**Pull request merged**